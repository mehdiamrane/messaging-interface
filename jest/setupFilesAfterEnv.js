import '@testing-library/jest-dom/extend-expect';

jest.mock('next-i18next', () => ({
  useTranslation: () => ({ t: (key) => key }),
}));
