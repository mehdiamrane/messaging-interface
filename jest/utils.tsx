import React from 'react';
import { render, type RenderOptions, type RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import theme from 'src/styles/theme';

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const AllProviders = ({ children }: { children: React.ReactNode }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

interface Render {
  (ui: React.ReactElement, options?: RenderOptions): RenderResult;
}

const customRender: Render = (ui, options) =>
  render(ui, {
    wrapper: (props) => <AllProviders {...props} />,
    ...options,
  });

export * from '@testing-library/react';
export { customRender as render };
