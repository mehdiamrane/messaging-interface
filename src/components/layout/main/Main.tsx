import React, { FC } from 'react';
import { StyledMain } from './Main.style';

type MainProps = {
  children: React.ReactNode;
  hideOnMobile?: boolean;
};

const Main: FC<MainProps> = ({ children, hideOnMobile }) => (
  <StyledMain.Container $hideOnMobile={hideOnMobile}>{children}</StyledMain.Container>
);

export default Main;
