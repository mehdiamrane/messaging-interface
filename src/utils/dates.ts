import formatDistanceToNow from 'date-fns/formatDistanceToNow';
import intlFormat from 'date-fns/intlFormat';
import { enUS, fr } from 'date-fns/locale';

const localeMapping = { 'en-US': enUS, fr };

export const formatRelativeDate = (date: Date, locale: string): string =>
  formatDistanceToNow(date, { locale: localeMapping[locale], addSuffix: true });

export const formatDate = (date: Date, locale: string): string =>
  intlFormat(
    date,
    {
      weekday: 'long',
      year: 'numeric',
      month: 'long',
      day: 'numeric',
      hour: 'numeric',
      minute: 'numeric',
    },
    {
      locale,
    },
  );

export const timestampToDate = (timestamp: number): Date => new Date(timestamp * 1000);
