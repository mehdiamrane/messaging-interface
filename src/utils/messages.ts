import type { Message } from 'src/types/message';
import type { ChatBubbleProps } from 'src/components/conversation/chatBubble/ChatBubble';

export const sortMessages = (messages: Message[]): Message[] =>
  (messages || []).sort((a, b) => a.timestamp - b.timestamp);

export const getMessagePosition = (messages: Message[], index: number): ChatBubbleProps['position'] => {
  const previous = messages[index - 1];
  const current = messages[index];
  const next = messages[index + 1];
  if (previous?.authorId !== current.authorId && next?.authorId !== current.authorId) {
    return 'single';
  }
  if (previous?.authorId !== current.authorId && next?.authorId === current.authorId) {
    return 'first';
  }
  if (previous?.authorId === current.authorId && next?.authorId !== current.authorId) {
    return 'last';
  }
  return 'middle';
};
