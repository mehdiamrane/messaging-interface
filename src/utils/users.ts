import type { User } from 'src/types/user';

// Default way to use a logged user
export const getLoggedUserId = (): User['id'] => 1;

export const isLoggedUser = (userId: User['id']): boolean => userId === getLoggedUserId();

export const getUserData = (userId: User['id'], users: User[]): User | undefined =>
  users.find(({ id }) => id === userId);

export const getAvatarFromNickname = (nickname: User['nickname']): string =>
  `https://ui-avatars.com/api/?bold=true&size=54&background=CBD5E0&rounded=true&name=${nickname}`;
