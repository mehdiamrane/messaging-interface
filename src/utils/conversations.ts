import type { Conversation } from 'src/types/conversation';
import type { BareUser } from 'src/types/user';
import { getLoggedUserId } from './users';

export const sortConversationsByLastActive = (conversations: Conversation[]): Conversation[] =>
  conversations.sort((a, b) => b.lastMessageTimestamp - a.lastMessageTimestamp);

export const getOtherUserFromConversation = (conversation: Conversation): BareUser =>
  conversation.senderId === getLoggedUserId()
    ? { id: conversation.recipientId, nickname: conversation.recipientNickname }
    : { id: conversation.senderId, nickname: conversation.senderNickname };
