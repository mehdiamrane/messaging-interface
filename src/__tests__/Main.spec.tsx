import React from 'react';
import { render } from 'jest/utils';
import Main from 'src/components/layout/main/Main';
import 'jest-styled-components';

describe('Main', () => {
  it('should render correctly Main', () => {
    const { container } = render(<Main />);
    expect(container).toMatchSnapshot();
  });
});
