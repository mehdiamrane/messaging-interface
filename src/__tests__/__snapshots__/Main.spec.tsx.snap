// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render correctly Main 1`] = `
.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
  min-height: calc(100% - 4rem - 4rem);
  width: 100%;
}

.c1 {
  margin: 0 auto;
  max-width: 72rem;
  padding: 2rem 1rem;
}

<div>
  <main
    class="c0"
  >
    <div
      class="c1"
    />
  </main>
</div>
`;
